mail/mailer.go

```diff
@@ -28,16 +49,58 @@ type MailerImpl struct {
 func New(server, port, username, password string) MailerImpl {
 	auth := smtp.PlainAuth("", username, password, server)
 	return MailerImpl{
-		Server: server,
-		Port:   port,
-		Auth:   auth,
-		From:   username,
+		Server:      server,
+		Port:        port,
+		Auth:        auth,
+		From:        username,
+		clk:         clock.Default(),
+		csprgSource: realSource{},
+	}
+}
+
+var maxBigInt = big.NewInt(math.MaxInt64) // ???
+
+func (m *MailerImpl) generateMessage(to []string, subject, body string) ([]byte, error) {
+	mid, err := m.csprgSource.Int(rand.Reader, maxBigInt)
+	if err != nil {
+		return nil, err
+	}
+	now := m.clk.Now().UTC()
+	addrs := []string{}
+	for _, a := range to {
+		addrs = append(addrs, strconv.QuoteToASCII(a))
```

Comment:
> Quote and QuoteToASCII convert strings to quoted Go string literals. The latter guarantees that the result is an ASCII string, by escaping any non-ASCII Unicode with \u

I don't think `\u` is a valid escape character per RFC822. Maybe you want instead to throw an error if there are non-ascii characters? Or pass them through in an appropriate way? Technically non-ASCII characters are allowed in the local part (left of the @) in email addresses. I haven't checked to see what we do with them in Boulder. I know many of the big US providers (hotmail, gmail) reject them. I'm sure there is someone with non-ASCII in their local part somewhere, but if we wanted to error on non-ASCII for now I would be okay with that.